toqito/matrix_ops/tests/test_tensor_comb.py

```diff
@@ -1,50 +1,67 @@
 """Test tensor_comb."""
-
 import numpy as np
 import pytest
 
-from toqito.matrix_ops import tensor_comb
-from toqito.states import basis
-
-e_0, e_1 = basis(2, 0), basis(2, 1)
+from toqito.matrix_ops import tensor_comb, to_density_matrix
 
 
 @pytest.mark.parametrize(
-    "states, k, expected",
+    "states, k, mode, density_matrix, expected_keys, expected_error",
     [
-        # Test tensor_comb for sequence length 1
-        ([e_0, e_1], 1, {(0,): np.outer(e_0, e_0), (1,): np.outer(e_1, e_1)}),
-        # Test tensor_comb for sequence length 2
-        (
-            [e_0, e_1],
-            2,
-            {
-                (0, 0): np.outer(np.kron(e_0, e_0), np.kron(e_0, e_0)),
-                (0, 1): np.outer(np.kron(e_0, e_1), np.kron(e_0, e_1)),
-                (1, 0): np.outer(np.kron(e_1, e_0), np.kron(e_1, e_0)),
-                (1, 1): np.outer(np.kron(e_1, e_1), np.kron(e_1, e_1)),
-            },
-        ),
-        # Test tensor_comb for sequence length 3
-        (
-            [e_0, e_1],
-            3,
-            {
-                (0, 0, 0): np.outer(np.kron(np.kron(e_0, e_0), e_0), np.kron(np.kron(e_0, e_0), e_0)),
-                (0, 0, 1): np.outer(np.kron(np.kron(e_0, e_0), e_1), np.kron(np.kron(e_0, e_0), e_1)),
-                # other possible sequences omitted for brevity...
-            },
-        ),
+        # Injective (default).
+        ([np.array([1, 0]), np.array([0, 1])],
+         3, "injective", False,
+         None,
+         ValueError),
+        # Invalid mode
+        ([np.array([1, 0]), np.array([0, 1])],
+         2, "invalid", False,
+         None,
+         ValueError),
+        # Injective mode with k = 1
+        ([np.array([1, 0]), np.array([0, 1])],
+         1, "injective", False,
+         [(0,), (1,)],
+         None),
+        # Non-injective mode with k = 1
+        ([np.array([1, 0]), np.array([0, 1])],
+         1, "non-injective", False,
+         [(0,), (1,)],
+         None),
+        # Non-injective mode with k = 2
+        ([np.array([1, 0]), np.array([0, 1])],
+         2, "non-injective", False,
+         [(0, 0), (0, 1), (1, 0), (1, 1)],
+         None),
+         # Non-injective mode with k = 3
+        ([np.array([1, 0]), np.array([0, 1])],
+         3, "non-injective", False,
+         [(0, 0, 0), (0, 0, 1)],
+         # other possible sequences omitted for brevity
+        None),
```

Comment:
Shouldn't this test that *all* of the sequences have been generated instead of just the first two?